rasa/core/lock.py

```diff
@@ -0,0 +1,160 @@
+import json
+import logging
+import typing
+from collections import deque
+from typing import Text, Optional, Union, Deque, Dict, Any
+
+import time
+
+if typing.TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+class Ticket:
+    def __init__(self, number: int, expires: float):
+        self.number = number
+        self.expires = expires
+
+    def has_expired(self):
+        return time.time() > self.expires
+
+    def dumps(self) -> Text:
+        """Return json dump of `Ticket`."""
+
+        return json.dumps(dict(number=self.number, expires=self.expires))
+
+    @classmethod
+    def from_dict(cls, data: Dict[Text, Any]) -> "Ticket":
+        """Creates `Ticket` from dictionary."""
+
+        return cls(data.get("number"), data.get("expires"))
```

Comment:
do we have to handle the case one of this dict values is `None`?

Reply:
how about parsing the number to number?

Reply:
no, they shouldn't be `None`.
> how about parsing the number to number?

do you mean cls(number=...)?

Reply:
nope, I guess `number` could also be a `str`, couldn't it? And do we have to handle the case if `data.get("number")` returns `None`?

Reply:
ah okay, I guess the dict typing is inaccurate then. `number` cannot be a string, as it's the issued ticket number (always an int). It can't be `None`, as all tickets have numbers

Reply:
ok, but  pytype might complain 😄

Reply:
no, pytype has been happy about this so far 😄

Reply:
updated: https://github.com/RasaHQ/rasa/pull/3933/files#diff-e1f07a9c6dc55a628ecd9e71a0c20a7dR26